app/javascript/src/components/Dashboard/Contacts/utils.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+
+import { MenuVertical } from "neetoicons";
+import { Avatar, Typography, Dropdown } from "neetoui";
+
+const renderAvatarWithNameRoleAndImage = (name, { role, imageUrl }) => (
+  <div className="flex flex-row items-center">
+    <Avatar
+      className="mr-2"
+      user={{
+        name,
+        imageUrl,
+      }}
+    />
+    <div>
+      <Typography style="h4">{name}</Typography>
+      <Typography style="body2">{role}</Typography>
+    </div>
+  </div>
+);
+
+const renderDropdownButton = (setShowDeleteAlert, setSelectedContactId, id) => (
+  <div>
+    <Dropdown buttonStyle="text" icon={MenuVertical}>
+      <li> Edit </li>
+      <li
+        onClick={() => {
+          setShowDeleteAlert(true);
+          setSelectedContactId(id);
+        }}
+      >
+        Delete
+      </li>
+    </Dropdown>
+  </div>
+);
+
+export const buildContactColumnData = (
+  setShowDeleteAlert,
+  setSelectedContactId
+) => [
+  {
+    dataIndex: "name",
+    key: "name",
+    title: "NAME & ROLE",
+
+    render: renderAvatarWithNameRoleAndImage,
+  },
+  {
+    dataIndex: "email",
+    key: "email",
+    title: "EMAIL",
+  },
+  {
+    dataIndex: "createdAt",
+    key: "createdAt",
+    title: "CREATED AT",
+
+    render: createdAt => (
+      <Typography style="body2">{createdAt.toDateString()}</Typography>
+    ),
+  },
+  {
+    dataIndex: "dropdown_menu",
+    key: "dropdown_menu",
+    title: "",
+
+    render: (item, { id }) =>
+      renderDropdownButton(setShowDeleteAlert, setSelectedContactId, id),
```

Comment:
Use named function parameters rather than positional.

Reply:
Could you explain why exactly named function parameters are preferred here over positional?

Reply:
@soorajbhskrn 

It's part of the [JavaScript RORO pattern](https://www.tinyblog.dev/blog/2020-07-13-javascript-roro-pattern/). Here, there are 3 positional parameters that you are required to pass whenever you are invoking `renderDropdownButton`. So this has few problems:
- Poor intellisense in VSCode. Intellisense always works better when arguments are named.
- You have to remember what item is at what position. So here, if you accidentally pass the second parameter as `id`, then the output will be different. With named arguments, order of arguments don't matter.
- Named arguments can allows destructing assignment. Example: `renderDropdownButton(buildRenderParams())`.
- The burden on developers to remember position of parameters is too much of a burden. Avoid it whenever possible. If there was only one argument, then we could prefer positional parameters.